### DISCUSSION TITLE
How to determine if a template has been deleted in gutenberg editor.

### DISCUSSION THREAD

**Discussion by weweloo** [WordPress/gutenberg#54525] [Q&A] [answered]
Use javascript, how to determine a template to be deleted in the full site editor.

**Comment by t-hamano** [comment, marked as answer]
Hi @weweloo,

Can you tell me specifically what you would like to achieve?

**Reply by weweloo** [reply]
I am using an additional database table to hold additional data for these templates, and when I delete a template in full site editor, this template's data in the database table needs to be deleted as well.

**Reply by t-hamano** [reply]
I see. Site editor templates are posts with a post type of wp_template. So, I think you can use the following core hook.

- [before_delete_post](https://developer.wordpress.org/reference/hooks/before_delete_post/)
- [delete_post](https://developer.wordpress.org/reference/hooks/delete_post/)
- [after_delete_post](https://developer.wordpress.org/reference/hooks/after_delete_post/)
